torchrl/objectives/sac.py

```diff
@@ -384,48 +393,54 @@ def __init__(
         if self._version == 1:
             self._vmap_qnetwork00 = vmap(qvalue_network)
 
+    @property
+    def target_entropy_buffer(self):
+        return self.target_entropy
+
     @property
     def target_entropy(self):
-        target_entropy = self.target_entropy_buffer
-        if target_entropy is None:
-            delattr(self, "target_entropy_buffer")
-            target_entropy = self._target_entropy
-            action_spec = self._action_spec
-            actor_network = self.actor_network
-            device = next(self.parameters()).device
-            if target_entropy == "auto":
-                action_spec = (
-                    action_spec
-                    if action_spec is not None
-                    else getattr(actor_network, "spec", None)
-                )
-                if action_spec is None:
-                    raise RuntimeError(
-                        "Cannot infer the dimensionality of the action. Consider providing "
-                        "the target entropy explicitely or provide the spec of the "
-                        "action tensor in the actor network."
-                    )
-                if not isinstance(action_spec, CompositeSpec):
-                    action_spec = CompositeSpec({self.tensor_keys.action: action_spec})
-                if (
-                    isinstance(self.tensor_keys.action, tuple)
-                    and len(self.tensor_keys.action) > 1
-                ):
-                    action_container_shape = action_spec[
-                        self.tensor_keys.action[:-1]
-                    ].shape
-                else:
-                    action_container_shape = action_spec.shape
-                target_entropy = -float(
-                    action_spec[self.tensor_keys.action]
-                    .shape[len(action_container_shape) :]
-                    .numel()
+        target_entropy = self._buffers.get("_target_entropy", None)
+        if target_entropy is not None:
+            return target_entropy
+        target_entropy = self._target_entropy
+        action_spec = self._action_spec
+        actor_network = self.actor_network
+        device = next(self.parameters()).device
+        if target_entropy == "auto":
+            action_spec = (
+                action_spec
+                if action_spec is not None
+                else getattr(actor_network, "spec", None)
+            )
+            if action_spec is None:
+                raise RuntimeError(
+                    "Cannot infer the dimensionality of the action. Consider providing "
+                    "the target entropy explicitely or provide the spec of the "
+                    "action tensor in the actor network."
```

Comment:
there is a tangential interesting bug related to this.

To see it try
```
from torchrl.modules import QValueActor, ProbabilisticActor, TanhDelta, ValueOperator
from tensordict.nn import TensorDictModule

from torchrl.objectives import SACLoss

if __name__ == "__main__":
    model = torch.nn.Linear(3, 4)  # does not work
    # model = torch.nn.Linear(1, 1) works
    actor_module = TensorDictModule(
        torch.nn.Linear(3, 4), in_keys=["obs"], out_keys=["logits"]
    )
    policy = ProbabilisticActor(
        module=actor_module,
        in_keys=["logits"],
        out_keys=["action"],
        distribution_class=TanhDelta,
    )
    value = ValueOperator(module=model, in_keys=["obs"], out_keys="value")

    loss = SACLoss(
        actor_network=policy,
        qvalue_network=value,
        # action_spec=UnboundedContinuousTensorSpec(shape=(2,)), not passing the spec
    )
    _ = loss.target_entropy
```
it will trigger this error but the user will see

```
File "/Users/matbet/PycharmProjects/rl/prova.py", line 29, in <module>
    loss.target_entropy
  File "/Users/matbet/PycharmProjects/rl/torchrl/objectives/common.py", line 348, in __getattr__
    return super().__getattr__(item)
  File "/Users/matbet/miniconda3/envs/torchrl/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1614, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'SACLoss' object has no attribute 'target_entropy'
```

so the message here is never shown

this is valid for all @property methods. isn't this curious?

Reply:
The `__getattr__` of torch.nn.Module isn't its greatest feature!
Let me think of a fix...

Reply:
the same extends also to `EnvBase` and other components, so the scope of this might be outside this PR